### PR TITLE
feat: Move csv writer to arrow CSV writer

### PR DESCRIPTION
**Pull request by yevgenypats**
Currently this use a [PR](https://github.com/apache/arrow/pull/34585) that is not upstream yet but should be soon so it's ready for review and we can roll this to our users. next will be parquet and then we can start moving this to SDK .

### CODE REVIEW

File: csv/write.go [hunk 1]
```diff
@@ -1,44 +1,68 @@
...
+	defer record.Release()
 
-func (cl *Client) WriteTableHeaders(w io.Writer, table *schema.Table) error {
-	writer := csv.NewWriter(w)
-	writer.Comma = cl.Delimiter
-
-	tableHeaders := make([]string, len(table.Columns))
-	for index, header := range table.Columns {
-		tableHeaders[index] = header.Name
+	writer := csv.NewWriter(w, arrowSchema, csv.WithComma(cl.Delimiter), csv.WithHeader(cl.IncludeHeaders))
```
Comment (hermanschaaf) [inline csv/write.go:26]: Wouldn't this write the headers every time a new batch is written?

Comment (yevgenypats) [reply csv/write.go:26]: If yes, then how it worked before? I don't think I changed something.

Comment (hermanschaaf) [reply csv/write.go:26]: Oh I see, we had a `WriteTableHeaders` function before, but it was being called by `WriteTableBatch` as well. So it should be fine, as long as a single batch maps to a single file

Comment (yevgenypats) [reply csv/write.go:26]: Yeah now the call to WriteTableHeaders is not needed as we use the writer and `csv.WithHeader`

**Comment by yevgenypats** [comment]
@hermanschaaf I tested now on the big 3: AWS, GCP, Azure and looks like format is the same at least per my manual test (I pushed a commit to make NULL an empty string to keep it backward compatible).

The only difference is the timestamp which I think this should be fine as the new timestamp is more accurate and always in UTC.

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**